BusinessLayer/SongControl/SongControl.cpp

```diff
@@ -21,7 +21,7 @@ QString SongControl::nextSong()
 
 QString SongControl::previousSong()
 {
-    currentSongIndex_ = (currentSongIndex_ - 1) % files_.size();
+    currentSongIndex_ = (((currentSongIndex_ - 1) % files_.size())+files_.size()) % files_.size();
```

Comment:
I think for readability, it'd just be easier if you did an `if` statement.
```
currentSongIndex_ -= 1;
if (currentSongIndex_ < 0) {
    currentSongIndex_ = files_.size() - 1;
}
```

Note: For future readers. C++ does not wrap negatives on modulus - I think...
https://stackoverflow.com/a/42964502